src/H5public.h

```diff
@@ -140,172 +106,41 @@ typedef int herr_t;
  *      printf("error determining whether data type is committed\n");
  *  }
  */
-#ifdef H5_HAVE_STDBOOL_H
-#include <stdbool.h>
-#else /* H5_HAVE_STDBOOL_H */
-#ifndef __cplusplus
-#if defined(H5_SIZEOF_BOOL) && (H5_SIZEOF_BOOL != 0)
-#define bool _Bool
-#else
-#define bool unsigned int
-#endif
-#define true 1
-#define false 0
-#endif /* __cplusplus */
-#endif /* H5_HAVE_STDBOOL_H */
```

Comment:
We should also change internal use of hbool_t to bool and ditch TRUE/FALSE in lieu of true/false.

Reply:
It's also entirely reasonable to retire hbool_t at this point.